x/evm/types/msg_evm.go

```diff
@@ -258,7 +259,20 @@ func (msg *MsgEthereumTx) firstVerifySig(chainID *big.Int) error {
 		return nil
 	}
 
-	V := new(big.Int)
+	var signer ethtypes.Signer
+	if isProtectedV(msg.Data.V) {
+		signer = ethtypes.NewEIP155Signer(chainID)
+	} else {
+		signer = ethtypes.HomesteadSigner{}
+	}
+
+	sigCache, ok := tmtypes.SignatureCache().Get(string(msg.TxHash()))
+	if ok && sigCache.EqualSiger(signer) {
```

Comment:
no need to check sigCache.EqualSiger(signer)

Reply:
> no need to check sigCache.EqualSiger(signer)

ok